connaisseur/validate.py

```diff
@@ -225,3 +201,38 @@ def search_image_targets_for_tag(trust_data: dict, image: Image):
 
     base64_digest = trust_data[image_tag]["hashes"]["sha256"]
     return base64.b64decode(base64_digest).hex()
+
+
+def update_with_delegation_trust_data(trust_data, delegations, key_store, host, image):
+    for delegation in delegations:
+        delegation_trust_data = get_delegation_trust_data(
+            host, image, TUFRole(delegation)
+        )
+        # when delegations are added to the repository, but weren't yet used for signing, the
+        # delegation files don't exist yet and are `None`. in this case validation must be skipped
+        if delegation_trust_data is not None:
+            delegation_trust_data.validate(key_store)
+        trust_data[delegation] = delegation_trust_data
+
+
+def validate_all_required_delegations_present(
```

Comment:
If this function is only used in this file, you may want to make it private (prepend with `_`)